### DISCUSSION TITLE
Burn speed now very slow after swapping out controllboard.

### DISCUSSION THREAD

**Discussion by maillemaker** [arkypita/LaserGRBL#1790] [Q&A] [answered]
I have an Eleksmaker A3 Pro.  It's control board is labeled "Mana SE".  The USB socket on the Arduino board has evidently come loose from the PCB, causing sporadic losses in connection.

I happened to have a spare Mana SE v3.2 board.  I have swapped it in, and it works, but the laser burn speed seems unaffected by the burn speed values inputted in LaserGrbl.

I was able to get the cable to persist its connection to the original board long enough to pull its configuration:
$0=10 (Step pulse time)
$1=25 (Step idle delay)
$2=0 (Step pulse invert)
$3=0 (Step direction invert)
$4=0 (Invert step enable pin)
$5=0 (Invert limit pins)
$6=0 (Invert probe pin)
$10=1 (Status report options)
$11=0.010 (Junction deviation)
$12=0.002 (Arc tolerance)
$13=0 (Report in inches)
$20=0 (Soft limits enable)
$21=0 (Hard limits enable)
$22=0 (Homing cycle enable)
$23=0 (Homing direction invert)
$24=100.000 (Homing locate feed rate)
$25=1000.000 (Homing search seek rate)
$26=250 (Homing switch debounce delay)
$27=1.000 (Homing switch pull-off distance)
$30=255 (Maximum spindle speed)
$31=0 (Minimum spindle speed)
$32=1 (Laser-mode enable)
$100=80.000 (X-axis travel resolution)
$101=80.000 (Y-axis travel resolution)
$102=80.000 (Z-axis travel resolution)
$110=4000.000 (X-axis maximum rate)
$111=4000.000 (Y-axis maximum rate)
$112=6000.000 (Z-axis maximum rate)
$120=800.000 (X-axis acceleration)
$121=800.000 (Y-axis acceleration)
$122=5000.000 (Z-axis acceleration)
$130=400.000 (X-axis maximum travel)
$131=300.000 (Y-axis maximum travel)
$132=200.000 (Z-axis maximum travel)

I have pushed this out to the new board.  The new boards' configuration was:
$0=10 (Step pulse time)
$1=25 (Step idle delay)
$2=0 (Step pulse invert)
$3=0 (Step direction invert)
$4=0 (Invert step enable pin)
$5=0 (Invert limit pins)
$6=0 (Invert probe pin)
$10=3 (Status report options)
$11=0.010 (Junction deviation)
$12=0.002 (Arc tolerance)
$13=0 (Report in inches)
$20=0 (Soft limits enable)
$21=0 (Hard limits enable)
$22=0 (Homing cycle enable)
$23=0 (Homing direction invert)
$24=25.000 (Homing locate feed rate)
$25=500.000 (Homing search seek rate)
$26=250 (Homing switch debounce delay)
$27=1.000 (Homing switch pull-off distance)
$100=80.000 (X-axis travel resolution)
$101=80.000 (Y-axis travel resolution)
$102=80.000 (Z-axis travel resolution)
$110=5000.000 (X-axis maximum rate)
$111=5000.000 (Y-axis maximum rate)
$112=5000.000 (Z-axis maximum rate)
$120=200.000 (X-axis acceleration)
$121=200.000 (Y-axis acceleration)
$122=200.000 (Z-axis acceleration)
$130=5000.000 (X-axis maximum travel)
$131=5000.000 (Y-axis maximum travel)
$132=5000.000 (Z-axis maximum travel)

When I try and push the old configuration to the new board, options 30, 31, and 32 do not go and generate an error.

I also tried flashing the Eleksmaker 3.8.hex back to the new board.  No effect.

**Comment by StuartB4** [comment, marked as answer]
Looks like the board needs flashing with GRBL 1.1
You can do that with LaserGRBL on the Tools menu.
V 1.1 includes the $30, 31 and 32 for laser mode. 

This is how. 

![Recovered_jpg_file(68)](https://user-images.githubusercontent.com/16980026/157346345-ef50b85e-cc98-4c0a-8e26-2d6fedf3e018.jpg)

![Recovered_jpg_file(69)](https://user-images.githubusercontent.com/16980026/157346356-36157227-ffe5-4f70-a373-6b9c847bef9f.jpg)

![Recovered_jpg_file(70)](https://user-images.githubusercontent.com/16980026/157346366-8f42c71b-fa2a-43f1-80de-9f2a87846f12.jpg)

![Recovered_jpg_file(71)](https://user-images.githubusercontent.com/16980026/157346375-581254c0-54c2-4966-a0c4-4d59a5054f1a.jpg)

![Recovered_jpg_file(72)](https://user-images.githubusercontent.com/16980026/157346389-ac4f34c3-2a0f-41bc-ba69-2d464506bd9f.jpg)

![Recovered_jpg_file(73)](https://user-images.githubusercontent.com/16980026/157346395-e891d9ff-741b-4814-aa64-acf3d0d743b3.jpg)

![Recovered_jpg_file(74)](https://user-images.githubusercontent.com/16980026/157346412-7423cd9f-249f-4613-ae45-9a5d69374892.jpg)

**Reply by maillemaker** [reply]
Indeed, the issue was that the firmware needed to be updated.  I found the EleksROM utility and used it to push 1.1 to the new board.  After than my old .nc file fully loaded properly, including options 30, 31, and 32.  Now the laser runs as it did before.  The EleksROM utility is evidently hard coded to provide only 2 .hex files and I can't specify the .h version which I have downloaded.  Now that you have showed me how to do it through LaserGrbl I may try that or, since it is now working, just leave it be.  :)

Thanks.

**Reply by StuartB4** [reply]
May be best to leave it if it's working ok now. 👍

**Reply by maillemaker** [reply]
Now I have another problem.  Machine burns for about 30 minutes, then the motor control goes wonky.  Laser is burning all over the place and the motors are stuttering as if they are jammed.  If I stop the burn and try to use the software motor controls I get stutter and not movement.  If I unplug the control board and plug it in again it seems to work for another 30 minutes.  I have upgraded to 1.1h via LaserGrble now.  I tried all the speed setting (ultra-fast down to low) didn't seem to help.

**Comment by StuartB4** [comment]
I would check the belts for tightness and the rollers and make sure there is nothing in the tracks.
Also the Metal pulleys on the motor shafts to make sure none of them are loose, and check for anything getting excessively hot.
Especially the motor driver chips on the main controller board. 
Check all the motor lead plugs and make sure they are all tight and you have no loose or broken wires in the plugs. 

Usually if you tighten the belts and push the laser all the way to the left and tip the machine up to about 45 degrees
to the right the laser should slide nice and smoothly to the other end without having to push it.
Same for the Y axis belts, bring it to the front and tip it up so the Y axis slides all the way to the other end, and make sure both sides of the axis reach the end at the same time. If one side gets there before the other the frame might not be properly square.

**Reply by maillemaker** [reply]
The unit worked fine with the previous Mana SE board, so I'm suspecting it is something with the board.

**Reply by maillemaker** [reply]
Figured it out.  Broken X axis servo cable connection.  Highly intermittent.  Basically when the gantry reached maximum distance away from the front of the machine, such that the cable was stretched out and tugging, it would break contact and then begin erratic operation.  Machine would operate fine on a diagonal burn pattern starting at the bottom left (zero), until it stretched out that cable far enough to break the contact.  Replaced the cable and all is well.

**Reply by StuartB4** [reply]
Thanks for letting us know. Good to know you found the problem. 👍
